lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+                            os.path.realpath(tb),
+                            tmpdir,
+                            Path(self.incoming, controller),
+                        )
+
+                        # Construct the generator for emitting all actions.  The
+                        # `idxctx` dictionary is passed along to each generator so
+                        # that it can add its context for error handling to the
+                        # list.
+                        idxctx.logger.debug("generator setup")
+                        if self.options.index_tool_data:
+                            actions = ptb.mk_tool_data_actions()
+                        else:
+                            actions = ptb.make_all_actions()
+
+                        # File name for containing all indexing errors that
+                        # can't/won't be retried.
+                        with ie_filepath.open(mode="w") as fp:
+                            idxctx.logger.debug("begin indexing")
+                            try:
+                                signal.signal(signal.SIGINT, sigint_handler)
+                                es_res = es_index(
+                                    idxctx.es, actions, fp, idxctx.logger, idxctx._dbg
+                                )
+                            except SigIntException:
+                                idxctx.logger.exception(
+                                    "Indexing interrupted by SIGINT, continuing to next tarball"
+                                )
+                                continue
+                            finally:
+                                # Turn off the SIGINT handler when not indexing.
+                                signal.signal(signal.SIGINT, signal.SIG_IGN)
+                    except UnsupportedTarballFormat as e:
+                        idxctx.logger.warning("Unsupported tar ball format: {}", e)
+                        tb_res = 4
+                    except BadDate as e:
+                        idxctx.logger.warning("Bad Date: {!r}", e)
+                        tb_res = 5
+                    except FileNotFoundError as e:
+                        idxctx.logger.warning("No such file: {}", e)
+                        tb_res = 6
+                    except BadMDLogFormat as e:
+                        idxctx.logger.warning(
+                            "The metadata.log file is curdled in" " tar ball: {}", e
+                        )
+                        tb_res = 7
+                    except SosreportHostname as e:
+                        idxctx.logger.warning("Bad hostname in sosreport: {}", e)
+                        tb_res = 10
+                    except tarfile.TarError as e:
+                        idxctx.logger.error(
+                            "Can't unpack tar ball into {}: {}", ptb.extracted_root, e
+                        )
+                        tb_res = 11
+                    except SigTermException:
```

Comment:
Do we need to handle SigTermException here? Although the log message will have to be generated elsewhere (presumbaly in `pbench-index.py` when the exception is handled).

Reply:
Mainly it is required here as we are handling sigquit in this block

Reply:
@ndokos, see the comment on lines 185 and 186. If there is a better way to word that comment to make this requirement clear, let me know; regardless, we should probably add a command like that here.